clusterloader2/pkg/framework/client/objects.go

```diff
@@ -110,20 +113,19 @@ func CreateObject(dynamicClient dynamic.Interface, namespace string, name string
 
 // UpdateObject updates with given name, group, version and kind based on given object description.
 func UpdateObject(dynamicClient dynamic.Interface, namespace string, name string, obj *unstructured.Unstructured) error {
-	// TODO(krzysied): This method should be more generic and robust.
 	gvk := obj.GroupVersionKind()
 	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
-	currentObj, err := GetObject(dynamicClient, gvk, namespace, name)
-	if err != nil {
-		return err
-	}
-	spec, exists := obj.UnstructuredContent()["spec"]
-	if !exists {
-		return fmt.Errorf("Object without a spec")
-	}
-	unstructured.SetNestedField(currentObj.UnstructuredContent(), spec, "spec")
+	obj.SetName(name)
 	updateFunc := func() error {
-		_, err := dynamicClient.Resource(gvr).Namespace(namespace).Update(currentObj)
+		currentObj, err := dynamicClient.Resource(gvr).Namespace(namespace).Get(name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		patch, err := createPatch(currentObj, obj)
+		if err != nil {
+			return fmt.Errorf("creating patch diff error: %v", err)
+		}
+		_, err = dynamicClient.Resource(gvr).Namespace(namespace).Patch(obj.GetName(), types.StrategicMergePatchType, patch)
```

Comment:
@krzysied @shyamjvs - this is no longer Update operation then. It should be called PatchObject or sth like that in that case.

Reply:
I agree. We should call this PatchObject based on what we're doing.

Also @krzysied, did you try using `Update()` instead? Did you face any problems with that?

Reply:
> Also @krzysied, did you try using Update() instead? Did you face any problems with that?

I've tried to use Update(). The problem was that it required to provided existing object with updated fields and I haven't found any support for this in k8s code.

So I've changed approach and I tried to implemented it the same way as `apply` is implemented in kubectl. It turned out `apply` is done by executing `patch` method.

> this is no longer Update operation then. It should be called PatchObject or sth like that in that case.

I have different opinion about this. I conciser patch as way of updating object - after executing this function the object will be (somehow) updated so the name seems valid.

I guess patch and update may have different meaning in k8s, so please let me know if that's the case.

Reply:
It's not about k8s - it's a more generic nomenclature. Yes - patching is a way of updating an object, but it's a different operation.
For Patch you provide a diff and that diff is applied to your object.
For Update, you provide a whole object to which the object should be updated.

I would still prefer using Update, but for now at the very least we should rename it to PatchObject.

Reply:
+1

> I would still prefer using Update

I thought about this a while ago and it wasn't very obvious to me how to generate updated object from current object and new json (for e.g how do we merge arrays?). That said, I guess we can continue to think about this with a TODO left for now.

Reply:
>For Patch you provide a diff and that diff is applied to your object.
For Update, you provide a whole object to which the object should be updated.

So if method takes whole object as a parameter, it should named 'update' according to your definition, right?

Reply:
What matters from our perspective (it's a framework for testing k8s) is what is send to k8s. And we are sending patch request to k8s.

Reply:
I've created #189 PR. It will change UpdateObject to PatchObject.